agroal-pool/src/main/java/io/agroal/pool/ConnectionHandler.java

```diff
@@ -94,6 +95,10 @@ public ConnectionHandler(XAConnection xa, Pool pool) throws SQLException {
         touch();
     }
 
+    public XAConnectionWrapper xaConnectionWrapper() {
+        return new XAConnectionWrapper( this, xaConnection, connectionPool.getConfiguration().connectionFactoryConfiguration().trackJdbcResources() );
```

Comment:
Why not add a getter for XAConnection to ConnectionHandler  like rawConnection and getXaResource and let wrapper class call it from handler like the other wrappers?

Reply:
that was an option, but don't want to expose the XAConnection if there is no need to do so